### PR TITLE
Update code to work with zig master

### PR DESCRIPTION
**Pull request by SamTebbs33**
This patch updates the codebase to work with latest zig master. The @noInlineCall builtin has been removed so I replaced the function calls in the mocking framework's `performAction` with normal function calls. Let me know if this needs to change.

### CODE REVIEW

File: test/mock/kernel/mock_framework.zig [hunk 1]
```diff
@@ -5,6 +5,8 @@ const expectEqual = std.testing.expectEqual;
 const GlobalAllocator = std.debug.global_allocator;
 const TailQueue = std.TailQueue;
 const warn = std.debug.warn;
+const builtin = @import("builtin");
+const Modifier = builtin.CallOptions.Modifier;
```
Comment (DrDeano) [inline test/mock/kernel/mock_framework.zig]: Is this needed as can't find it in the code

File: test/mock/kernel/idt_mock.zig [hunk 2]
```diff
@@ -39,7 +39,7 @@ const NUMBER_OF_ENTRIES: u16 = 256;
 const TABLE_SIZE: u16 = @sizeOf(IdtEntry) * NUMBER_OF_ENTRIES - 1;
 
 pub fn openInterruptGate(index: u8, handler: InterruptHandler) IdtError!void {
-    return mock_framework.performAction("openInterruptGate", IdtError!void, index, handler);
+    return mock_framework.performAction("openInterruptGate", IdtError!void, .{ index, handler });
```
Comment (DrDeano) [inline test/mock/kernel/idt_mock.zig]: Space at the end of the .{}? Is this what zig fmt outputs

File: test/mock/kernel/arch_mock.zig [hunk 3]
```diff
@@ -36,47 +36,47 @@ pub const InterruptContext = struct {
 };
 
 pub fn outb(port: u16, data: u8) void {
-    return mock_framework.performAction("outb", void, port, data);
+    return mock_framework.performAction("outb", void, .{ port, data });
```
Comment (DrDeano) [inline test/mock/kernel/arch_mock.zig]: Same here, spaces

File: src/kernel/tty.zig [hunk 4]
```diff
@@ -439,10 +439,10 @@ fn printCallback(ctx: void, str: []const u8) TtyError!void {
 ///     IN comptime format: []const u8 - The format string to print
 ///     IN args: ...                   - The arguments to be used in the formatted string
 ///
-pub fn print(comptime format: []const u8, args: ...) void {
+pub fn print(comptime format: []const u8, args: var) void {
```
Comment (DrDeano) [inline src/kernel/tty.zig]: Need to update the doc comments of functions that have the new `var` type. Could also check the other files as well :)

File: src/kernel/log.zig [hunk 5]
```diff
@@ -12,38 +12,38 @@ fn logCallback(context: void, str: []const u8) anyerror!void {
     serial.writeBytes(str, serial.Port.COM1);
 }
 
-pub fn log(comptime level: Level, comptime format: []const u8, args: ...) void {
+pub fn log(comptime level: Level, comptime format: []const u8, args: var) void {
```
Comment (DrDeano) [inline src/kernel/log.zig]: There are no doc comments on the logging functions

File: test/mock/kernel/mock_framework.zig (hunk 1 above)
Comment (SamTebbs33) [reply test/mock/kernel/mock_framework.zig]: It certainly isn't. It remains from when I was looking into replacing @noInlineCall

File: test/mock/kernel/idt_mock.zig (hunk 2 above)
Comment (SamTebbs33) [reply test/mock/kernel/idt_mock.zig]: It is

File: src/kernel/tty.zig (hunk 4 above)
Comment (SamTebbs33) [reply src/kernel/tty.zig]: Well spotted

File: test/mock/kernel/idt_mock.zig (hunk 2 above)
Comment (DrDeano) [reply test/mock/kernel/idt_mock.zig]: When I run zig fmt, it removes the spaces.

Comment (SamTebbs33) [reply test/mock/kernel/idt_mock.zig]: Hmm it doesn't when I try. What command are you running?

Comment (DrDeano) [reply test/mock/kernel/idt_mock.zig]: Found that `zig fmt` is inconsistent with formatting the `.{}`. So will fix later

**Pull request merged**